src/__tests__/request_middleware.test.ts

```diff
@@ -107,3 +111,70 @@ describe("withRequestMiddlewares", () => {
     });
   });
 });
+
+describe("withRequestMiddlewaresT", () => {
```

Comment:
I'm not sure this scenario is covered by the implementation, but it might require a dedicated test:

when executing a tuple `t` of n middlewares `t=[m1,m2,...mn]`, the handler should be called with n parameters, each well-typed according with the type of the middlware.

Reply:
I added another test to check that types are respected in [2a1aa7f](https://github.com/pagopa/io-ts-commons/pull/370/commits/2a1aa7fe9ff0fe7ef5f26bea55266beed99e80ce)
There is a commented piece of code that shows a not-woking configuration, because of type mismatch.

Tests are not working yet in code review pipelines

Reply:
@balanza I made tests work by disabling ts-jest diagnostic options. Don't like it at all, let me know if you have a better solution

Reply:
We can use [`// @ts-expect-error` annotation](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments) to make test assertion on assignments for which we expect TS to throw an error

Reply:
Thanks, I've added a new test in [e0caf40](https://github.com/pagopa/io-ts-commons/pull/370/commits/e0caf40def847de7cee0cf466adb50c14670c8ee)

To bypass the error I was experiencing in code-review pipeline I added an ignored code in jest configuration [here](https://github.com/pagopa/io-ts-commons/pull/370/commits/239c9f275a35c5acd3e74f88c96f10fb93530caa)